paasta_tools/secret_providers/vault.py

```diff
@@ -140,3 +142,36 @@ def decrypt_secret(self, secret_name: str) -> str:
             cache_dir=None,
             context=self.service_name,
         ).decode('utf-8')
+
+    def decrypt_secret_raw(self, secret_name: str) -> bytes:
+        ecosystem = self.get_vault_ecosystems_for_clusters()[0]
+        secret_path = os.path.join(
+            self.secret_dir,
+            f"{secret_name}.json",
+        )
+        if not hasattr(self, 'client'):
```

Comment:
Is there a reason why you're doing this `hasattr()` check and not usual `self.client = None` in `__init__()` and then `if self.client is not None:` here?

Reply:
ug, yeah I wasn't super happy with this either. I think it was something like when decrypting we only connect to 1 cluster but encrypting we talk to multiple. Anyway, let me re-visit and see if I can come up with something less ugly.

Reply:
ok, just pushed a commit to clean this up a little!